src/valintapiste_service/pistetiedot.clj

```diff
@@ -105,3 +107,26 @@
                                            (doseq [row rows]
                                                (upsert-valintapiste! tx (pistetieto-row-for-update row))))
                                          conflicting-hakemus-oids))] data)))
+
+(defn create-siirtotiedostot-for-pistetiedot
+  "Create siirtotiedosto containing pistetiedot hakemuksittain"
+  [datasource siirtotiedosto-client start-datetime end-datetime max-hakemuscount-in-file]
+  (if (> max-hakemuscount-in-file 32767) (throw (IllegalArgumentException. (str "Illegal value " max-hakemuscount-in-file " for number of hakemukset per file, max number is 32767!"))))
+  (let [connection {:datasource datasource}
+        rows (jdbc/with-db-transaction [tx connection]
+                                               (if-not (nil? start-datetime)
+                                                 (find-hakemus-oids-by-timerange tx {:start start-datetime :end end-datetime})
```

Comment:
Onko jokin syy miksi haetaan ensin oidit ja sitten pisteet? Ne näyttävät tulevat samasta taulusta. Jos haettaisiin samalla niin tarvittaisiinko partitiointia?

Reply:
Ei mitään varsinaista syytä. Voitaisiin tosiaan toteuttaa käyttäen yläpuolella mainittua possu-logiikkaa, tällöin päästäisiin eroon partitioista ja maagisesta 32767:stä.